prow/plugins/verify-owners/verify-owners.go

```diff
@@ -0,0 +1,146 @@
+	LoadRepoOwners(org, repo, base string) (repoowners.RepoOwnerInterface, error)
+}
+
+type githubClient interface {
+	AddLabel(org, repo string, number int, label string) error
+	CreateReview(org, repo string, number int, r github.DraftReview) error
+	GetPullRequestChanges(org, repo string, number int) ([]github.PullRequestChange, error)
+}
+
+func handlePullRequest(pc plugins.PluginClient, pre github.PullRequestEvent) error {
+	if pre.Action != github.PullRequestActionOpened && pre.Action != github.PullRequestActionReopened && pre.Action != github.PullRequestActionSynchronize {
+		return nil
+	}
+	return handle(pc.GitHubClient, pc.GitClient, pc.Logger, &pre, pc.PluginConfig.Owners.LabelsBlackList)
+}
+
+func handle(ghc githubClient, gc *git.Client, log *logrus.Entry, pre *github.PullRequestEvent, labelsBlackList []string) error {
+	org := pre.Repo.Owner.Login
+	repo := pre.Repo.Name
+	var errFile string
+	var errMsg error
+
+	// Get changes.
+	changes, err := ghc.GetPullRequestChanges(org, repo, pre.Number)
+	if err != nil {
+		return fmt.Errorf("error getting PR changes: %v", err)
+	}
+
+	// List modified OWNERS files.
+	var modifiedOwnersFiles []string
+	for _, change := range changes {
+		if filepath.Base(change.Filename) == ownersFileName {
+			modifiedOwnersFiles = append(modifiedOwnersFiles, change.Filename)
+		}
+	}
+	if len(modifiedOwnersFiles) == 0 {
+		return nil
+	}
+
+	// Clone the repo, checkout the PR.
+	r, err := gc.Clone(pre.Repo.FullName)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := r.Clean(); err != nil {
+			log.WithError(err).Error("Error cleaning up repo.")
+		}
+	}()
+	if err := r.CheckoutPullRequest(pre.Number); err != nil {
+		return err
+	}
+
+	// Check each OWNERS file.
+	for _, f := range modifiedOwnersFiles {
+		// Try to load OWNERS file.
+		simple, err := repoowners.ParseSimpleConfig(filepath.Join(r.Dir, f))
+		if err != nil {
+			errFile = f
+			errMsg = fmt.Errorf("error occurred parsing the OWNERS file in %s: %v", errFile, err)
+			break
+		}
+		// Check labels against blacklist
+		if sets.NewString(simple.Config.Labels...).HasAny(labelsBlackList...) {
+			errMsg = fmt.Errorf("OWNERS file contains blacklisted labels: %v", sets.NewString(simple.Config.Labels...).Intersection(sets.NewString(labelsBlackList...)))
+			break
+		}
+		// Check approvers isn't empty
+		if filepath.Dir(f) == "." && len(simple.Config.Approvers) == 0 {
+			errMsg = errors.New("no approvers defined for the repo")
+			break
+		}
+	}
+	// React if we saw something.
+	if errMsg != nil {
+		log.Infof("Adding %s label.", invalidOwnersLabel)
+		if err := ghc.AddLabel(org, repo, pre.Number, invalidOwnersLabel); err != nil {
+			return err
+		}
+		resp := fmt.Sprintf("Your OWNERS file is invalid: %v", errMsg)
+		log.Infof("Creating a review with \"%s\".", resp)
+		ghc.CreateReview(org, repo, pre.Number, github.DraftReview{
+			Action: github.Comment,
+			Comments: append([]github.DraftReviewComment{}, github.DraftReviewComment{
```

Comment:
I think it would be less visually cluttered to make one review with all of the comments, see the `lint` plugin for an example.